src/CommitmentsV2/SFA.DAS.CommitmentsV2/Application/Queries/GetApprenticeships/GetApprenticeshipsHandler.cs

```diff
@@ -26,24 +28,253 @@ public class GetApprenticeshipsHandler : IRequestHandler<GetApprenticeshipsReque
+            }
+            else if (string.IsNullOrEmpty(request.SortField) && request.ReverseSort)
+            {
+                searchResult = await ApprenticeshipsByReverseDefaultOrder(cancellationToken, request.ProviderId, request.PageNumber, request.PageItemCount);
+            }
+            else
+            {
+                searchResult = await ApprenticeshipsByDefaultOrder(cancellationToken, request.ProviderId, request.PageNumber, request.PageItemCount);
+            }
+
+            foreach (var apprenticeship in searchResult.Apprenticeships)
+            {
+                var details = await _mapper.Map(apprenticeship);
+                matchedApprenticeshipDetails.Add(details);
+            }
+
+            return new GetApprenticeshipsResponse
+            {
+                Apprenticeships = matchedApprenticeshipDetails,
+                TotalApprenticeshipsFound = searchResult.TotalApprenticeshipsFound,
+                TotalApprenticeshipsWithAlertsFound = searchResult.TotalApprenticeshipsWithAlertsFound
+            };
+        }
+
+        private async Task<ApprenticeshipSearchResult> ApprenticeshipsByDefaultOrder(CancellationToken cancellationToken, long? providerId, int pageNumber, int pageItemCount)
+        {
+            var apprenticeships = await _dbContext
+                .Apprenticeships
+                .Where(apprenticeship => apprenticeship.Cohort.ProviderId == providerId && apprenticeship.PendingUpdateOriginator != null)
+                .OrderBy(x => x.FirstName)
+                .ThenBy(x => x.LastName)
+                .ThenBy(x => x.Uln)
+                .ThenBy(x => x.Cohort.LegalEntityName)
+                .ThenBy(x => x.CourseName)
+                .ThenByDescending(x => x.StartDate)
+                .Include(apprenticeship => apprenticeship.Cohort)
+                .Include(apprenticeship => apprenticeship.DataLockStatus)
+                .ToListAsync(cancellationToken);
+            
+            var apprenticeshipsWithoutAlerts = await _dbContext
+                .Apprenticeships
+                .Where(apprenticeship => apprenticeship.Cohort.ProviderId == providerId && apprenticeship.PendingUpdateOriginator == null)
+                .OrderBy(x => x.FirstName)
+                .ThenBy(x => x.LastName)
+                .ThenBy(x => x.Uln)
+                .ThenBy(x => x.Cohort.LegalEntityName)
+                .ThenBy(x => x.CourseName)
+                .ThenByDescending(x => x.StartDate)
                 .Include(apprenticeship => apprenticeship.Cohort)
                 .Include(apprenticeship => apprenticeship.DataLockStatus)
-                .Where(apprenticeship => apprenticeship.Cohort.ProviderId == request.ProviderId)
                 .ToListAsync(cancellationToken);
 
-            foreach (var apprenticeship in matched)
+            var totalApprenticeshipsWithAlertsFound = apprenticeships.Count;
+
+            apprenticeships.AddRange(apprenticeshipsWithoutAlerts);
```

Comment:
Is there no way of writing this without performing two queries and addranging them?